log-analyzer/features/org.wso2.carbon.la.apim.feature/src/main/resources/gadgets/LogAPIMArtifactDeleted/js/main.js

```diff
@@ -0,0 +1,144 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+var client = new AnalyticsClient().init();
+var div = "#tblArtifactDeleted";
+var table;
+var from = new Date(moment().subtract(1, 'year')).getTime();
+var to = new Date(moment()).getTime();
+var dataM = [];
+var initState = true;
+
+var meta = {
+    "names": ["apiArtifact", "Frequency"],
+    "types": ["ordinal", "linear"]
+};
+
+var configTable = {
+    key: "apiArtifact",
+    title:"LogAPIMArtifact",
+    charts: [{
+        type: "table",
+        y: "Frequency",
+        columns: ["apiArtifact", "Frequency"],
+        columnTitles: ["APIM Artifact", "Frequency"]
+    }
+    ],
+    width: $('body').width(),
+    height: $('body').height(),
+    padding: { "top": 40, "left": 80, "bottom": 70, "right": 100 }
+};
+
+
+function initialize() {
+    fetch();
+    //$("#tblArtifactDeleted").html(getDefaultText());
+}
+
+function getDefaultText() {
+    return '<div class="status-message">'+
+        '<div class="message message-info">'+
+        '<h4><i class="icon fw fw-info"></i>No content to display</h4>'+
+        '<p>Please select a date range to view stats.</p>'+
+        '</div>'+
+        '</div>';
+};
+
+function getEmptyRecordsText() {
+    return '<div class="status-message">'+
+        '<div class="message message-info">'+
+        '<h4><i class="icon fw fw-info"></i>No records found</h4>'+
+        '<p>Please select a date range to view stats.</p>'+
+        '</div>'+
+        '</div>';
+}
+
+$(document).ready(function () {
+    initialize();
+});
+
+function fetch() {
+    dataM.length = 0;
+    var queryInfo;
+    console.log("sajith3");
+    queryInfo = {
+        tableName: "LOGANALYZER_APIM_ARTIFACT_DELETED_DAILY",
+        searchParams: {
+            query: "_timestamp: [" + from + " TO " + to + "]",
+            start : 0, //starting index of the matching record set
+            count : 100 //page size for pagination
+        }
+    };
+    console.log(queryInfo);
+    client.search(queryInfo, function (d) {
+        var obj = JSON.parse(d["message"]);
+        if (d["status"] === "success") {
+            for (var i =0; i < obj.length ;i++){
+                dataM.push([obj[i].values.artifact,obj[i].values.artifactCount]);
+            }
+            if(!initState){
+                redrawLogAPIMArtifactTableChart();
+            }else{
+                drawLogAPIMArtifactTableChart();
+                initState =false;
+            }
+        }
+    }, function (error) {
+        console.log("error occured: " + error);
```

Comment:
Logging in the console doesn't notify the user at all. This error needs to be propagated to the front-end and then the error needs to be notified to the user.